src/js/Icon.js

```diff
@@ -1,31 +1,39 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import deepAssign from 'deep-assign';
-import { CHANNEL } from 'styled-components/lib/models/ThemeProvider';
+import {
+  CHANNEL_NEXT,
+  CONTEXT_CHANNEL_SHAPE,
+} from 'styled-components/lib/models/ThemeProvider';
 
 import StyledIcon from './StyledIcon';
 
 class Icon extends Component {
   static contextTypes = {
     grommet: PropTypes.object,
     theme: PropTypes.object,
-    [CHANNEL]: PropTypes.func,
+    [CHANNEL_NEXT]: CONTEXT_CHANNEL_SHAPE,
   };
 
   state = {
     theme: undefined,
   };
 
+  unsubscribeId: -1;
+
   componentWillMount() {
-    const subscribe = this.context[CHANNEL];
-    if (typeof subscribe === 'function') {
-      this.unsubscribe = subscribe(theme => this.setState({ theme }));
+    const styledContext = this.context[CHANNEL_NEXT];
+    if (styledContext !== undefined) {
+      const { subscribe } = styledContext;
+      this.unsubscriberId = subscribe(theme => this.setState({ theme }));
     }
   }
 
   componentWillUnmount() {
-    if (typeof this.unsubscribe === 'function') {
-      this.unsubscribe();
+    const styledContext = this.context[CHANNEL_NEXT];
+    if (styledContext !== undefined && this.unsubscribeId !== -1) {
```

Comment:
I believe it is fine to check just for the subscription id here, since it will only be initialized if styled component is in the context.

```
if (this. scSubscriptionId) {
}
```

Reply:
Will check this.